### PR TITLE
Add specs for reconfiguring vm with OvirtSDK

### PR DESCRIPTION
**Pull request by borod108**
Add some specs to the reconfiguring vm process

### CODE REVIEW

**Comment by borod108** [comment]
@masayag please take a look

**Comment by borod108** [comment]
@masayag can you please review?

File: spec/models/manageiq/providers/redhat/infra_manager_spec.rb [hunk 1]
```diff
@@ -43,97 +43,133 @@
...
+            @ems.vm_reconfigure(@vm)
+          end
+        end
 
-      expect(@rhevm_vm).to receive(:cpu_topology=).with(:cores => @cores_per_socket, :sockets => @num_of_sockets)
-      @ems.vm_reconfigure(@vm, :spec => spec)
+        context "use_ovirt_engine_sdk is set to false" do
+          let(:use_ovirt_engine_sdk) { false }
+          it 'sends vm_reconfigure to the right ovirt_services' do
+            expect(@v4_strategy_instance).to receive(:vm_reconfigure).with(@vm, {})
```
Comment (masayag) [inline spec/models/manageiq/providers/redhat/infra_manager_spec.rb]: this seems incorrect: if sdk shouldn't be used, the v4 strategy instance shouldn't be used.

File: spec/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4_spec.rb [hunk 2]
```diff
@@ -70,4 +70,111 @@
...
+          let(:spec) { { 'memoryMB' => 8.gigabytes / 1.megabyte + 1 } }
+          let(:memory) { 8.gigabytes + 256.megabytes }
+          it 'adjusts the increased memory to the next 256 MiB multiple if the VM is up' do
+            expect(@vm_service).to receive(:update).with(OvirtSDK4::Vm.new(@memory_spec), :next_run => true)
+            expect(@vm_service).to receive(:update).with(OvirtSDK4::Vm.new(:memory => memory))
+            reconfigure_vm
+          end
+        end
+
+        context "memory is less than vms memory should be rounded up to" do
```
Comment (masayag) [inline spec/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4_spec.rb]: should there be any text after "up do" ? this seems like the sentence is ended prematurely.

**Comment by masayag** [comment]
Tests look good, added few comments inline.

For v4 we support also reconfigure of vm's disks. Those aren't covered by the tests.
This can be added later on or we can settle with the exiting disks tests coverage we already have.

File: spec/models/manageiq/providers/redhat/infra_manager_spec.rb (hunk 1 above)
Comment (borod108) [inline spec/models/manageiq/providers/redhat/infra_manager_spec.rb]: It should. The vm_reconfigure should always run through v4 if it is available even when the use_ovirt_sdk is toggled off, this is by design.

File: spec/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4_spec.rb (hunk 2 above)
Comment (borod108) [inline spec/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4_spec.rb]: done

**Comment by borod108** [comment]
@masayag if I fixed your comments properly and you do not have any more, can you approve this and assign oourfali?

**Comment by simaishi** [comment]
Fine backport details:

```
$ git log -1
commit e3114c4cfe39787f1192650df56b4251a879d52a
Author: Oved Ourfali <oourfali@redhat.com>
Date:   Tue May 23 13:41:45 2017 +0300

    Merge pull request #22 from borod108/specs/reconfigure_vm
    
    Add specs for reconfiguring vm with OvirtSDK
    (cherry picked from commit ed7a2da288aeeabed8a73e39e130e131eb724d73)
```

**Pull request merged**